From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jake Bailey <5341706+jakebailey@users.noreply.github.com>
Date: Tue, 4 Oct 2022 17:01:38 -0700
Subject: [PATCH] Remove all files in lib before LKG

---
 scripts/produceLKG.mjs | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/scripts/produceLKG.mjs b/scripts/produceLKG.mjs
index 9b2c4e7d1e..0d1ea0f8bc 100644
--- a/scripts/produceLKG.mjs
+++ b/scripts/produceLKG.mjs
@@ -1,8 +1,8 @@
-import childProcess from "child_process";
 import fs from "fs-extra";
 import path from "path";
 import glob from "glob";
 import url from "url";
+import del from "del";
 
 const __filename = url.fileURLToPath(new URL(import.meta.url));
 const __dirname = path.dirname(__filename);
@@ -14,6 +14,8 @@ const copyright = fs.readFileSync(path.join(__dirname, "../CopyrightNotice.txt")
 
 async function produceLKG() {
     console.log(`Building LKG from ${source} to ${dest}`);
+    await del(`${dest.replace(/\\/g, "/")}/**`, { ignore: ["**/README.md"] });
+    await fs.mkdirp(dest);
     await copyLibFiles();
     await copyLocalizedDiagnostics();
     await copyTypesMap();
@@ -94,16 +96,6 @@ async function copyFilesWithGlob(pattern) {
     console.log(`Copied ${files.length} files matching pattern ${pattern}`);
 }
 
-/**
- * @param {string} path
- * @param {string[]} args
- */
-async function exec(path, args = []) {
-    const cmdLine = ["node", path, ...args].join(" ");
-    console.log(cmdLine);
-    childProcess.execSync(cmdLine);
-}
-
 process.on("unhandledRejection", err => {
     throw err;
 });
-- 
2.38.1

