From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jake Bailey <5341706+jakebailey@users.noreply.github.com>
Date: Tue, 20 Sep 2022 13:54:37 -0700
Subject: [PATCH] Try and fix up LKG

---
 Gulpfile.js           | 18 ++++++------------
 scripts/produceLKG.ts | 43 ++++++++++++++++++++++++++++++++-----------
 2 files changed, 38 insertions(+), 23 deletions(-)

diff --git a/Gulpfile.js b/Gulpfile.js
index 7079bd0298..54949534e6 100644
--- a/Gulpfile.js
+++ b/Gulpfile.js
@@ -561,27 +561,20 @@ const importDefinitelyTypedTests = () => exec(process.execPath, ["scripts/import
 task("importDefinitelyTypedTests", series(buildImportDefinitelyTypedTests, importDefinitelyTypedTests));
 task("importDefinitelyTypedTests").description = "Runs the importDefinitelyTypedTests script to copy DT's tests to the TS-internal RWC tests";
 
-// TODO(jakebailey): There isn't a release build anymore; figure out what to do here.
-// Probably just use tsc.js.
-
-const buildReleaseTsc = () => buildProject("src/tsc/tsconfig.release.json");
-const cleanReleaseTsc = () => cleanProject("src/tsc/tsconfig.release.json");
-cleanTasks.push(cleanReleaseTsc);
-
 const cleanBuilt = () => del("built");
 
 const produceLKG = async () => {
+    // TODO(jakebailey): there are probably more files here that are needed.
     const expectedFiles = [
-        "built/local/tsc.release.js",
-        "built/local/typescriptServices.js",
-        "built/local/typescriptServices.d.ts",
+        "built/local/tsc.js",
         "built/local/tsserver.js",
         "built/local/typescript.js",
         "built/local/typescript.d.ts",
         "built/local/tsserverlibrary.js",
         "built/local/tsserverlibrary.d.ts",
         "built/local/typingsInstaller.js",
-        "built/local/cancellationToken.js"
+        "built/local/cancellationToken.js",
+        "built/local/watchGuard.js",
     ].concat(libs.map(lib => lib.target));
     const missingFiles = expectedFiles
         .concat(localizationTargets)
@@ -597,7 +590,8 @@ const produceLKG = async () => {
     }
 };
 
-task("LKG", series(lkgPreBuild, parallel(localize, buildTsc, buildServer, buildServices, buildLssl, buildOtherOutputs, buildReleaseTsc), produceLKG));
+// TODO(jakebailey): dependencies on dts
+task("LKG", series(lkgPreBuild, parallel(localize, buildTsc, buildServer, buildServices, buildLssl, buildOtherOutputs), produceLKG));
 task("LKG").description = "Makes a new LKG out of the built js files";
 task("LKG").flags = {
     "   --built": "Compile using the built version of the compiler.",
diff --git a/scripts/produceLKG.ts b/scripts/produceLKG.ts
index 23a553631c..1df84ed324 100644
--- a/scripts/produceLKG.ts
+++ b/scripts/produceLKG.ts
@@ -28,6 +28,27 @@ async function copyLocalizedDiagnostics() {
     const dir = await fs.readdir(source);
     const ignoredFolders = ["enu"];
 
+    // TODO(jakebailey): Instead of ignoring folders, we should keep a list of
+    // the localizationTargets somewhere that can be used by multiple modules.
+    ignoredFolders.push(
+        "compiler",
+        "deprecatedCompat",
+        "executeCommandLine",
+        "harness",
+        "jsTyping",
+        "loggedIO",
+        "server",
+        "services",
+        "testRunner",
+        "tsc",
+        "tsserver",
+        "tsserverlibrary",
+        "typescript",
+        "typingsInstaller",
+        "typingsInstallerCore",
+        "webServer",
+    );
+
     for (const d of dir) {
         const fileName = path.join(source, d);
         if (
@@ -44,20 +65,20 @@ async function copyTypesMap() {
 }
 
 async function copyScriptOutputs() {
-    await copyWithCopyright("cancellationToken.js");
-    await copyWithCopyright("tsc.release.js", "tsc.js");
-    await copyWithCopyright("tsserver.js");
-    await copyFromBuiltLocal("tsserverlibrary.js"); // copyright added by build
-    await copyFromBuiltLocal("typescript.js"); // copyright added by build
-    await copyFromBuiltLocal("typescriptServices.js"); // copyright added by build
-    await copyWithCopyright("typingsInstaller.js");
-    await copyWithCopyright("watchGuard.js");
+    // TODO(jakebailey): This does not work when unbundled.
+    // TODO(jakebailey): Copyright is added by esbuild; maybe we should do it here?
+    await copyFromBuiltLocal("cancellationToken.js");
+    await copyFromBuiltLocal("tsc.js");
+    await copyFromBuiltLocal("tsserver.js");
+    await copyFromBuiltLocal("tsserverlibrary.js");
+    await copyFromBuiltLocal("typescript.js");
+    await copyFromBuiltLocal("typingsInstaller.js");
+    await copyFromBuiltLocal("watchGuard.js");
 }
 
 async function copyDeclarationOutputs() {
-    await copyFromBuiltLocal("tsserverlibrary.d.ts"); // copyright added by build
-    await copyFromBuiltLocal("typescript.d.ts"); // copyright added by build
-    await copyFromBuiltLocal("typescriptServices.d.ts"); // copyright added by build
+    await copyWithCopyright("tsserverlibrary.d.ts");
+    await copyWithCopyright("typescript.d.ts");
 }
 
 async function writeGitAttributes() {
-- 
2.37.3

