From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jake Bailey <5341706+jakebailey@users.noreply.github.com>
Date: Tue, 4 Oct 2022 17:01:38 -0700
Subject: [PATCH] Pull LKG script change from other branch

---
 scripts/produceLKG.ts | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/scripts/produceLKG.ts b/scripts/produceLKG.ts
index 9ce5bcc63f..9da846a3be 100644
--- a/scripts/produceLKG.ts
+++ b/scripts/produceLKG.ts
@@ -3,6 +3,7 @@
 import * as fs from "fs-extra";
 import * as path from "path";
 import * as glob from "glob";
+import * as del from "del";
 
 const root = path.join(__dirname, "..");
 const source = path.join(root, "built/local");
@@ -10,6 +11,8 @@ const dest = path.join(root, "lib");
 
 async function produceLKG() {
     console.log(`Building LKG from ${source} to ${dest}`);
+    await del(`${dest.replace(/\\/g, "/")}/**`, { ignore: ["**/README.md"] });
+    await fs.mkdirp(dest);
     await copyLibFiles();
     await copyLocalizedDiagnostics();
     await copyTypesMap();
-- 
2.38.0

