From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jake Bailey <5341706+jakebailey@users.noreply.github.com>
Date: Tue, 4 Oct 2022 21:56:58 -0700
Subject: [PATCH] Temporarily disable template literals

---
 Gulpfile.mjs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Gulpfile.mjs b/Gulpfile.mjs
index 22f9446898..5ab3802d95 100644
--- a/Gulpfile.mjs
+++ b/Gulpfile.mjs
@@ -175,6 +175,7 @@ function esbuildTask(entrypoint, outfile, exportIsTsObject = false) {
         supported: {
             // "const-and-let": false, // https://github.com/evanw/esbuild/issues/297
             "object-rest-spread": false, // Performance enhancement, see: https://github.com/evanw/esbuild/releases/tag/v0.14.46
+            "template-literal": false, // TODO(jakebailey): workaround for https://github.com/microsoft/TypeScript/issues/51072
         },
         // legalComments: "none", // If we add copyright headers to the source files, uncomment.
         plugins: [
-- 
2.38.0

