From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jake Bailey <5341706+jakebailey@users.noreply.github.com>
Date: Fri, 30 Sep 2022 16:19:43 -0700
Subject: [PATCH] Add ts to globalThis for convenience during debugging

---
 src/testRunner/runner.ts | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/testRunner/runner.ts b/src/testRunner/runner.ts
index 28bfd96983..528a3f6930 100644
--- a/src/testRunner/runner.ts
+++ b/src/testRunner/runner.ts
@@ -280,6 +280,9 @@ function beginTests() {
 
 export let isWorker: boolean;
 function startTestEnvironment() {
+    // For debugging convenience.
+    (globalThis as any).ts = require("./_namespaces/ts");
+
     isWorker = handleTestConfig();
     if (isWorker) {
         return Parallel.Worker.start();
-- 
2.37.3

