From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jake Bailey <5341706+jakebailey@users.noreply.github.com>
Date: Tue, 18 Oct 2022 19:49:16 -0700
Subject: [PATCH] Remove Promise redeclaration, our target includes Promise

---
 src/services/utilities.ts | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/src/services/utilities.ts b/src/services/utilities.ts
index fa38c0dab0..be9740a77e 100644
--- a/src/services/utilities.ts
+++ b/src/services/utilities.ts
@@ -60,24 +60,6 @@ import {
     unescapeLeadingUnderscores, UserPreferences, VariableDeclaration, visitEachChild, VoidExpression, YieldExpression,
 } from "./_namespaces/ts";
 
-/** @internal */ // Don't expose that we use this
-// Based on lib.es6.d.ts
-declare global {
-    // Module transform: converted from ambient declaration
-    /** @internal */
-    interface PromiseConstructor {
-        new <T>(executor: (resolve: (value?: T | PromiseLike<T>) => void, reject: (reason?: any) => void) => void): Promise<T>;
-        reject(reason: any): Promise<never>;
-        all<T>(values: (T | PromiseLike<T>)[]): Promise<T[]>;
-    }
-}
-/** @internal */
-declare global {
-    // Module transform: converted from ambient declaration
-    /** @internal */
-    var Promise: PromiseConstructor; // eslint-disable-line no-var
-}
-
 // These utilities are common to multiple language service features.
 //#region
 /** @internal */
-- 
2.38.1

