From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jake Bailey <5341706+jakebailey@users.noreply.github.com>
Date: Thu, 13 Oct 2022 11:05:42 -0700
Subject: [PATCH] Fix tests and add explainer back

---
 Herebyfile.mjs | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Herebyfile.mjs b/Herebyfile.mjs
index 57760e8fc9..000bec4c1b 100644
--- a/Herebyfile.mjs
+++ b/Herebyfile.mjs
@@ -195,7 +195,12 @@ async function runDtsBundler(entrypoint, output) {
         target: "es2018", // Covers Node 10.
         format: "cjs",
         sourcemap: "linked",
+        // external/conditions are workarounds for: https://github.com/evanw/esbuild/issues/1958
+        // For requires in the compiler/server/etc, we have to ensure to write things indirectly
+        // to avoid hardcoding a path to node_modules. In the test code, we can be less careful,
+        // but we need to ensure we import the CJS versions of the libraries, hence setting conditions.
         external: ["./node_modules/*"],
+        conditions: ["require"],
         supported: {
             // "const-and-let": false, // https://github.com/evanw/esbuild/issues/297
             "object-rest-spread": false, // Performance enhancement, see: https://github.com/evanw/esbuild/releases/tag/v0.14.46
-- 
2.38.0

