From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jake Bailey <5341706+jakebailey@users.noreply.github.com>
Date: Tue, 18 Oct 2022 21:45:07 -0700
Subject: [PATCH] Add note about tsserver bundle format

---
 Herebyfile.mjs | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Herebyfile.mjs b/Herebyfile.mjs
index 72dd08e9e1..7913984824 100644
--- a/Herebyfile.mjs
+++ b/Herebyfile.mjs
@@ -369,6 +369,10 @@ const { main: server } = entrypointBuildTask({
     builtEntrypoint: "./built/local/tsserver/server.js",
     output: "./built/local/tsserver.js",
     mainDeps: [generateLibs],
+    // Even though this seems like an exectuable, so could be the default CJS,
+    // this is used in the browser too. Do the same thing that we do for our
+    // libraries and generate an IIFE with name `ts`, as to not pollute the global
+    // scope.
     esbuildOptions: { exportIsTsObject: true },
 });
 export { server };
-- 
2.38.0

